pandas/tests/tseries/test_timezones.py

```diff
@@ -65,6 +64,145 @@ def cmptz(self, tz1, tz2):
+        assert self.cmptz(ts2.index.tz, self.tz('US/Eastern'))
+
+    def test_ambiguous_bool(self):
+        # make sure that we are correctly accepting bool values as ambiguous
+
+        # GH#14402
+        ts = Timestamp('2015-11-01 01:00:03')
+        expected0 = Timestamp('2015-11-01 01:00:03-0500', tz='US/Central')
+        expected1 = Timestamp('2015-11-01 01:00:03-0600', tz='US/Central')
+
+        ser = Series([ts])
+        expected0 = Series([expected0])
+        expected1 = Series([expected1])
+
+        with pytest.raises(pytz.AmbiguousTimeError):
+            ser.dt.tz_localize('US/Central')
+
+        result = ser.dt.tz_localize('US/Central', ambiguous=True)
+        assert_series_equal(result, expected0)
+
+        result = ser.dt.tz_localize('US/Central', ambiguous=[True])
+        assert_series_equal(result, expected0)
+
+        result = ser.dt.tz_localize('US/Central', ambiguous=False)
+        assert_series_equal(result, expected1)
+
+        result = ser.dt.tz_localize('US/Central', ambiguous=[False])
+        assert_series_equal(result, expected1)
+
+    def test_localized_at_time_between_time(self):
+        from datetime import time
+
+        rng = date_range('4/16/2012', '5/1/2012', freq='H')
+        ts = Series(np.random.randn(len(rng)), index=rng)
+
+        ts_local = ts.tz_localize(self.tzstr('US/Eastern'))
+
+        result = ts_local.at_time(time(10, 0))
+        expected = ts.at_time(time(10, 0)).tz_localize(self.tzstr(
+            'US/Eastern'))
+        assert_series_equal(result, expected)
+        assert self.cmptz(result.index.tz, self.tz('US/Eastern'))
+
+        t1, t2 = time(10, 0), time(11, 0)
+        result = ts_local.between_time(t1, t2)
+        expected = ts.between_time(t1,
+                                   t2).tz_localize(self.tzstr('US/Eastern'))
+        assert_series_equal(result, expected)
+        assert self.cmptz(result.index.tz, self.tz('US/Eastern'))
+
+    def test_string_index_alias_tz_aware(self):
+        rng = date_range('1/1/2000', periods=10, tz=self.tzstr('US/Eastern'))
+        ts = Series(np.random.randn(len(rng)), index=rng)
+
+        result = ts['1/3/2000']
+        tm.assert_almost_equal(result, ts[2])
+
+    def test_tz_aware_asfreq(self):
+        dr = date_range('2011-12-01', '2012-07-20', freq='D',
+                        tz=self.tzstr('US/Eastern'))
+
+        s = Series(np.random.randn(len(dr)), index=dr)
+
+        # it works!
+        s.asfreq('T')
+
+    def test_dateutil_tzoffset_support(self):
+        values = [188.5, 328.25]
+        tzinfo = tzoffset(None, 7200)
+        index = [datetime(2012, 5, 11, 11, tzinfo=tzinfo),
+                 datetime(2012, 5, 11, 12, tzinfo=tzinfo)]
+        series = Series(data=values, index=index)
+
+        assert series.index.tz == tzinfo
+
+        # it works! #2443
+        repr(series.index[0])
+
+    def test_getitem_pydatetime_tz(self):
+        index = date_range(start='2012-12-24 16:00', end='2012-12-24 18:00',
```

Comment:
indexing test

Reply:
This test uses methods of this class; let's be extra careful before breaking it up.